adapters/adquery/adquery.go

```diff
@@ -0,0 +1,211 @@
+package adquery
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const (
+	defaultCurrency string = "PLN"
+	bidderName      string = "adquery"
+	prebidVersion   string = "server"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Adquery adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) buildRequest(bidReq *openrtb2.BidRequest, imp *openrtb2.Imp, ext *openrtb_ext.ImpExtAdQuery) *BidderRequest {
+	userId := ""
+	if bidReq.User != nil {
+		userId = bidReq.User.ID
+	}
+
+	return &BidderRequest{
+		V:                   prebidVersion,
+		PlacementCode:       ext.PlacementID,
+		AuctionId:           "",
+		BidType:             ext.Type,
+		AdUnitCode:          imp.TagID,
+		BidQid:              userId,
+		BidId:               fmt.Sprintf("%s%s", bidReq.ID, imp.ID),
+		Bidder:              bidderName,
+		BidderRequestId:     bidReq.ID,
+		BidRequestsCount:    1,
+		BidderRequestsCount: 1,
+		Sizes:               getImpSizes(imp),
+	}
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if len(request.Imp) == 0 {
+		return nil, nil
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	var result []*adapters.RequestData
+	var errs []error
+	for _, imp := range request.Imp {
+		ext, err := parseExt(imp.Ext)
+		if err != nil {
+			errs = append(errs, &errortypes.BadInput{err.Error()})
+			continue
+		}
+
+		requestJSON, err := json.Marshal(a.buildRequest(request, &imp, ext))
+		if err != nil {
+			return nil, []error{err}
```

Comment:
`return nil, []error{err}` this will override errors previously appended (line 72) in `errs`. Instead should change line 78 to append marshal error

```
   requestJSON, err := json.Marshal(a.buildRequest(request, &imp, ext))
   if err !=nil {
      return nil, append(errs, err)
   }
```